x/delayedack/rollapp_hooks.go

```diff
@@ -120,6 +120,18 @@ func (im IBCMiddleware) onRecvPacket(rollappPacket commontypes.RollappPacket, lo
 		if err != nil {
 			return
 		}
+		/*
+			Here, we do the inverse of what we did when we updated the packet transfer address, when we fulfilled the order
+			to ensure the ack matches what the rollapp expects.
+		*/
+		transferPacketData, err := rollappPacket.GetTransferPacketData()
+		if err != nil {
+			return fmt.Errorf("get transfer packet data: %w", err)
+		}
+		if rollappPacket.OriginalRecvReceiver != "" {
```

Comment:
I'd add a comment on why it can be empty here for clarity

Reply:
added